File: sdk/python/mashgate/resources/storage.py
"""Storage resource (mgStorage)."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from mashgate.client import MashgateClient


class StorageResource:
    def __init__(self, client: MashgateClient) -> None:
        self._c = client

    def generate_upload_url(
        self,
        *,
        tenant_id: str,
        filename: str,
        content_type: str | None = None,
        max_size_bytes: int | None = None,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"tenantId": tenant_id, "filename": filename}
        if content_type is not None:
            body["contentType"] = content_type
        if max_size_bytes is not None:
            body["maxSizeBytes"] = max_size_bytes
        return self._c.request("POST", "/v1/storage/upload-url", body=body)

    def list_files(self, tenant_id: str) -> dict[str, Any]:
        return self._c.request("GET", "/v1/storage/files", query={"tenantId": tenant_id})

    def delete_file(self, file_id: str, *, tenant_id: str) -> None:
        self._c.request(
            "DELETE", f"/v1/storage/files/{file_id}", query={"tenantId": tenant_id}
        )

    def get_download_url(self, file_id: str, *, tenant_id: str) -> dict[str, Any]:
        return self._c.request(
            "GET", f"/v1/storage/files/{file_id}/download", query={"tenantId": tenant_id}
        )

File: sdk/typescript/src/resources/storage.ts
import type { MashgateClient } from "../client.js";
import type { FileEntry, UploadUrlResponse } from "../types.js";

export interface GenerateUploadUrlRequest {
  tenantId: string;
  filename: string;
  mimeType: string;
}

export class StorageResource {
  constructor(private readonly client: MashgateClient) {}

  async generateUploadUrl(data: GenerateUploadUrlRequest): Promise<UploadUrlResponse> {
    return this.client.request<UploadUrlResponse>("POST", "/v1/storage/upload-url", {
      body: data,
    });
  }

  async listFiles(tenantId: string): Promise<FileEntry[]> {
    return this.client.request<FileEntry[]>("GET", "/v1/storage/files", {
      query: { tenantId },
    });
  }

  async deleteFile(fileId: string, tenantId: string): Promise<void> {
    return this.client.request<void>("DELETE", `/v1/storage/files/${fileId}`, {
      query: { tenantId },
    });
  }

  async getDownloadUrl(fileId: string, tenantId: string): Promise<string> {
    const result = await this.client.request<{ url: string }>(
      "GET",
      `/v1/storage/files/${fileId}/url`,
      { query: { tenantId } },
    );
    return result.url;
  }
}
